### PR TITLE
add correct types for classes inherited from SimpleSAMLphp

### PR DESCRIPTION
**Pull request by briskt**
### Changed (breaking)
- Updated `process` method typehinting in `Auth\ProcessingFilter` inherited classes
- Updated `login` and `authenticated` method types in `SilAuth`
- Updated `log` and `validateIdP` method types in `IdPDisco`
- Added types to `FakeSession` class inherited from `SimpleSAML\Session`

To keep this pull request relatively small in scope, tests have been skipped because they are not yet passing.

[Reference](https://simplesamlphp.org/docs/stable/simplesamlphp-upgrade-notes-2.0.html#changes-relevant-for-module-developers)

https://itse.youtrack.cloud/issue/IDP-900

### CODE REVIEW

File: modules/sildisco/src/Auth/Process/LogUser.php [hunk 1]
```diff
@@ -65,9 +65,9 @@ public function __construct(array $config, mixed $reserved)
     /**
      * Log info for a user's login to Dynamodb
      *
-     * @param array &$state  The current state array
```
Comment (jason-jackson) [inline modules/sildisco/src/Auth/Process/LogUser.php:68]:
```suggestion

```

File: modules/sildisco/src/Auth/Process/TagGroup.php [hunk 2]
```diff
@@ -31,9 +31,10 @@ public function prependIdp2Groups(array $attributes, string $attributeLabel, str
     /**
      * Apply filter to copy attributes.
      *
-     * @param array &$state  The current request
```
Comment (jason-jackson) [inline modules/sildisco/src/Auth/Process/TagGroup.php:34]:
```suggestion
```

File: modules/sildisco/src/Auth/Process/TrackIdps.php [hunk 3]
```diff
@@ -11,9 +11,10 @@ class TrackIdps extends \SimpleSAML\Auth\ProcessingFilter {
     /**
      * Apply filter to save IDPs to session.
      *
-     * @param array &$request  The current request
```
Comment (jason-jackson) [inline modules/sildisco/src/Auth/Process/TrackIdps.php:14]:
```suggestion
```

**Review by jason-jackson — APPROVED** [review]

File: modules/sildisco/src/Auth/Process/LogUser.php (hunk 1 above)
Comment (briskt) [reply modules/sildisco/src/Auth/Process/LogUser.php:68]: The "Log info for a user's login to Dynamodb" description is for this class, not the Auth/Process class. Combining this phrase with @inheritDoc provides both descriptions in the function documentation, as best I understand.

Comment (jason-jackson) [reply modules/sildisco/src/Auth/Process/LogUser.php:68]: Ah, ok, thanks!

**Review by jason-jackson — APPROVED** [review]

**Pull request merged**